Public/Src/Cache/ContentStore/Distributed/NuCache/ContentLocationDatabase.cs

```diff
@@ -616,15 +627,26 @@ private Task ForceCacheFlushAsync(OperationContext context)
                 Tracer,
                 async () =>
                 {
+                    long flushedEntries = 0;
                     try
                     {
+                        // Make it likely that this runs in a separate thread other than the one that triggered the
+                        // flush
                         await Task.Yield();

```

Comment:
I believe this Task.Yield should be inside '_inMemoryCache.FlushAsync' not here.

Right now it is hard to tell, why the Yield is here. THis method just do its job and it is hard to tell why we follow this pattern here (calling Yield) but not in other places.

Moving this into InMemoryCache.FlushAsync make sense, because InMemoryCache.FlushAsync is not "very" async because it only acquires mutex asynchronously.